code/modules/mod/mod_control.dm

```diff
@@ -694,12 +693,13 @@
 		return
 	atom_destruction(damage_flag)
 
-/obj/item/mod/control/proc/on_part_deletion(obj/item/part) //the part doesnt count as being qdeleted, so our destroying does an infinite loop, fix later
+/obj/item/mod/control/proc/on_part_deletion(obj/item/part)
 	SIGNAL_HANDLER
 
 	if(QDELETED(src))
 		return
 	qdel(src)
+	return TRUE
```

Comment:
why the return?

Reply:
Returns TRUE to COMSIG_PARENT_PREQDELETED, stopping qdel loop and lets the suit handle deletion of all the parts.

Reply:
this should probably be its own signal return define, but since it isnt, add a comment please